bulk-model-sync-lib/src/jvmTest/kotlin/org/modelix/model/sync/bulk/ModelImporterOrderPropertyTest.kt

```diff
@@ -0,0 +1,355 @@
+                  {
+                    "id": "child2",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  },
+                  {
+                    "id": "child4",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  },
+                  {
+                    "id": "child1",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  }
+                ]
+              }
+            ]
+          }
+        }
+        """.trimIndent()
+        val dataToImport = ModelData.fromJson(dataToImportJson)
+        modelImporter.import(dataToImport)
+
+        val children = branch.getRootNode().allChildren.first().allChildren.toList()
+        // child1 and child2 should not be reordered because they have an unordered role
+        // child3 is removed
+        // child4 is added at the end
+        assertEquals(listOf("child1", "child2", "child4"), children.map(INode::originalId))
+    }
+
+    @Test
+    fun `unordered children with changing parents are appended`() {
+        // language=json
+        val dataBeforeImportJson = """
+        {
+          "root": {
+            "children": [
+              {
+                "id": "parent1",
+                "concept": "uid1",
+                "children": [
+                  {
+                    "id": "child1",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  },
+                  {
+                    "id": "child2",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  }
+                ]
+              }
+            ]
+          }
+        }
+        """.trimIndent()
+        val dataBeforeImport = ModelData.fromJson(dataBeforeImportJson)
+        val branch = loadModelIntoBranch(dataBeforeImport)
+        val modelImporter = ModelImporter(branch.getRootNode())
+
+        // language=json
+        val dataToImportJson = """
+        {
+          "root": {
+            "children": [
+              {
+                "id": "parent1",
+                "concept": "uid1",
+                "children": [
+                  {
+                    "id": "child1",
+                    "concept": "uid1",
+                    "role": "uid3"
+                  },
+                  {
+                    "id": "child2",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  }
+                ]
+              }
+            ]
+          }
+        }
+        """.trimIndent()
+        val dataToImport = ModelData.fromJson(dataToImportJson)
+        modelImporter.import(dataToImport)
+
+        val children = branch.getRootNode().allChildren.first().allChildren.toList()
+        // child1 moved with because of the role change from "uid2" to "uid3".
+        assertEquals(listOf("child2", "child1"), children.map(INode::originalId))
```

Comment:
It's not valid to assume that `.allChildren` returns nodes in different roles are in any specific order. These tests probably shouldn't use `.allChildren` at all.

Reply:
Done. (`.allChildren` is now only used to navigate from the root to the relevant parents.)